software-mansion/react-native-reanimated
# Add version property to worklets for Reanimated plugin version verification

## Summary

This PR modifies the way we verify if there is no mismatch between versions of Reanimated and Reanimated Babel plugin (for eg. when metro cache was not reset).

- Added `__version` prop to every worklet in DEV.
- Added version checking in runtime.
- Added global variable to ensure error for the mismatch is logged only once.
- Removed old `checkPluginVersion.ts` and moved `checkCppVersion.ts` back to `checkVersion.ts`.
- Changed plugin's code and jest tests accordingly.

## Test plan

0. Run `yarn jest`.
1. Make a DEV bundle to confirm `__version` prop appears on worklets.
2. Make a production to bundle to confirm it doesn't appear there.
3. Build ExampleApp and fiddle with `jsVersion.ts` to confirm that it's working.
